example/poisson_gmg/poisson_driver.cpp

```diff
@@ -292,9 +304,12 @@ TaskCollection PoissonDriver::MakeTaskCollectionMG(BlockList_t &blocks) {
   }
 
   for (int ivcycle = 0; ivcycle < max_iterations; ++ivcycle) {
-    TaskRegion &region = tc.AddRegion(num_partitions * (max_level + 1));
-    for (int level = max_level; level >= min_level; --level) {
-      AddMultiGridTasksLevel(region, level, min_level, max_level, level == min_level);
+    int mg_reg_dep_id = 0;
+    TaskRegion &region = tc.AddRegion(num_partitions);
+    for (int i = 0; i < num_partitions; ++i) {
+      TaskList &tl = region[i];
+      AddMultiGridTasksLevel(region, tl, none, i, mg_reg_dep_id, max_level, min_level,
+                             max_level);
     }
 
```

Comment:
I had just copied what was done in some other code for this where the `reg_dep_id` was initialized outside of the loop over partitions and then incremented for each partition. It is not clear to me how the regional dependence ids work under the hood, so maybe it is not correct as is. The desired behavior is to have "all blocks in each partition being logically in the same region so that all blocks on the mesh progress region by region".

Reply:
In that case, I suggest to defer the question to @jdolence who IIRC wrote the regional dependencies so that we don't run into unexpected issues with `pack_size != 1` or `!= -1`